### PR TITLE
Allow to overwrite a single language string via the theme folder

### PR DESCRIPTION
**Pull request by nickvergessen**
Anti whitespace: https://github.com/owncloud/core/pull/22677/files?w=1
### Steps
1. Use a theme
2. Switch the language to "Deutsch (Persönlich)"
3. Put the file `core/l10n/de.js` with the following content into your theme:

``` js
OC.L10N.register(
  "core",
  {
    "Share link" : "Niemand sollte diese Checkbox auswählen"
  },
  "nplurals=2; plural=(n != 1);"
);
```
1. Open the sharing sidebar
### Expected

German page with "Niemand sollte diese Checkbox auswählen" checkbox
### Actual

Everything is reset to english, apart from the "Niemand sollte diese Checkbox auswählen" checkbox

The problem is we only loaded one l10n resource, so when we found the one in the theme we didn't load the core version with the remaining translations anymore.
Now we load all languages (first core, then theme) and overwrite core strings with the theme.

Fixes https://github.com/owncloud/enterprise/issues/1109

cc @MorrisJobke @DeepDiver1975 @PVince81

### CODE REVIEW

**Comment by nickvergessen** [comment]
@karlitschek could backport this to 8.2 to make life easier

**Comment by karlitschek** [comment]
makes sense. please backport

**Comment by PVince81** [comment]
Tested, works :+1:

File: core/js/l10n.js [hunk 1]
```diff
@@ -66,13 +66,21 @@ OC.L10N = {
...
+			} else {
+				// generate plural function based on form
+				this._pluralFunctions[appName] = this._generatePluralFunction(pluralForm);
+			}
 		} else {
-			// generate plural function based on form
-			this._pluralFunctions[appName] = this._generatePluralFunction(pluralForm);
+			// Theme overwriting the default language
+			_.each(bundle, function(translation, key) {
+				self._bundles[appName][key] = translation
```
Comment (PVince81) [inline core/js/l10n.js]: Missing semicolon.

Note: it should also be possible to do this using `self._bundles[appName] = _.extend(self._bundles[appName], bundle)` which would copy all keys from `bundle` into the other one.

**Comment by PVince81** [comment]
@nickvergessen https://github.com/owncloud/core/commit/0091df2bc8eed432df9d726d5bea09ac527128d2

**Comment by nickvergessen** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Pull request merged**